commit 3db0cb4107a7c609e3614bd448fec321916fafa7 (refs/changes/99/3899/3)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-05-17T09:51:53+02:00 (1 year, 5 months ago)
    
    TOOLS-1224 'sdcadm <subcommand> -h|--help' hits SAPI to get the sdc app: that's overkill
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/cli/index.js b/lib/cli/index.js
index e5bd67c..9871d66 100644
--- a/lib/cli/index.js
+++ b/lib/cli/index.js
@@ -102,12 +102,14 @@ CLI.prototype.init = function init(opts, args, callback) {
         logComponent = handler.name.slice(3);
     }
     var logToFile = (handler && handler.logToFile || false);
+    var skipInit = false;
 
     if (args.indexOf('help') !== -1 ||
         args.indexOf('--help') !== -1 ||
         args.indexOf('-h') !== -1 ||
         opts.help) {
         logToFile = false;
+        skipInit = true;
     }
 
     // Wrap into try/catch block and handle ENOSPC and EACCES with friendlier
@@ -180,7 +182,7 @@ CLI.prototype.init = function init(opts, args, callback) {
          * manually skip for those here.
          */
         var skipInitCmds = {completion: true, help: true};
-        if (args[0] && skipInitCmds[args[0]]) {
+        if (skipInit || (args[0] && skipInitCmds[args[0]])) {
             callback();
             return;
         }
